app/vlinsert/insertutil/line_reader.go

```diff
@@ -69,9 +68,36 @@ func (lr *LineReader) NextLine() bool {
 		}
 
 		buf := lr.buf[lr.bufOffset:]
-		if n := bytes.IndexByte(buf, '\n'); n >= 0 {
-			lr.Line = buf[:n]
-			lr.bufOffset += n + 1
+		var size uint64
+		if readSize {
```

Comment:
TBH this change looks irrelevant. If line already has `\n` delimiter, there is no need to read any kind of size. 

 However, if this size symbol breaks parsing logic on caller side, just add a simple check for it at journald.go. 

 Also, please add documentation or RFC link if logic covers some uneven corner. And it also be great to have a test for it.